### ISSUE TITLE
enh(skore): Enable `data_source="all"` in metrics accessor

### ISSUE THREAD

**Issue by sylvaincom** [probabl-ai/skore#1446]
### Is your feature request related to a problem? Please describe.

As a data scientist, I would like to see if my model is overfitting or not, by comparing the score of my model on the train set and on the test set.

As of v0.9.1, to get the summary of metrics on the test set, I can do:
```python
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split
from sklearn.linear_model import Ridge
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler
from skore import EstimatorReport

X, y = fetch_california_housing(return_X_y=True, as_frame=True)

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

ridge_report = EstimatorReport(
    make_pipeline(StandardScaler(), Ridge()),
    X_train=X_train,
    X_test=X_test,
    y_train=y_train,
    y_test=y_test,
)

ridge_report.metrics.summarize().frame()  # default on test
```
and for the train:
```python
ridge_report.metrics.summarize(data_source="train").frame() 
```
But I can't have both the train and test sets on the same dataframe.

### Describe the solution you'd like

Have something like:
```python
ridge_report.metrics.summarize(data_source="all").frame()
```
for both train and test sets, which would return this kind of dataframe:

![Image](https://github.com/user-attachments/assets/f2dab250-c375-4fc2-a224-ba1e31d72575)

**Comment by glemaitre** [comment]
Agreed. It is particularly useful when we want to show the effect of underfitting or overfitting.
I assume that there is over places where we want to check those.

**Comment by avinabneogy23** [comment]
Hey @MarieSacksick , can you please assign this issue to me

**Comment by aryan0931** [comment]
can I work on this issue

**Comment by MarieSacksick** [comment]
hello @aryan0931, it seems that @avinabneogy23 is working on it currently; let's let them a couple of weeks before changing the assignee! If on May 16th they didn't have time, I guess you can start doing a PR.
Thanks!

**Comment by MangKaGO** [comment]
can i be asigned to this task ?

**Comment by sylvaincom** [comment]
I updated this issue following:
- https://github.com/probabl-ai/skore/pull/1813
- https://github.com/probabl-ai/skore/pull/1825

**Comment by MatthieuLoustaunau** [comment]
I can contribute to this issue (Saint-Gobain workshop)
